### PR TITLE
V2.2- voyage Plan Tests

### PR DESCRIPTION
**Pull request by JayanthiiB**
### Notes

---

**Remember to merge to feature branches not `master`.**

### CODE REVIEW

File: cypress/integration/sGMR/add-people.spec.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ describe('Add People in account', () => {
         cy.get('.govuk-error-message').each((error,index) =>{
             cy.wrap(error).should('contain.text',errors[index]).and('be.visible');
         })
-        cy.enterPeopleInfo1(people);
```
Comment (p3dr0migue1) [inline cypress/integration/sGMR/add-people.spec.js:39]: Where is `enterPeopleInfo1` coming from? What is replacing it?

File: cypress/integration/sGMR/add-vessel.spec.js [hunk 2]
```diff
@@ -34,14 +33,17 @@ describe('Add new pleasure craft in account', () => {
...
         cy.get('#pleasureCraftNameInput').clear().type("NewVesselName");
-        //need change?
         cy.get('.govuk-radios__item').eq(0).click();
         cy.get('#mooringInput').clear().type("Moorings123");
         cy.contains('Continue').click();
         const errors2 =[
             'You must enter a registration number',
             'You must select if your pleasure craft has a country of registration',
-            'You must select if the pleasure craft has a call sign',
+        'You must select if the pleasure craft has an Automatic Identification System (AIS)',
```
Comment (p3dr0migue1) [inline cypress/integration/sGMR/add-vessel.spec.js:42]: Please mind indentation

File: cypress/integration/sGMR/add-voyage-report-with-saved-data.spec.js [hunk 3]
```diff
@@ -10,7 +10,7 @@ describe('Add voyage plan with saved data', () => {
     let vessel;
     let persons = [];
     let departDate;
-    const numberOfPersons = 3;
+    const numberOfPersons = 2;
```
Comment (p3dr0migue1) [inline cypress/integration/sGMR/add-voyage-report-with-saved-data.spec.js:13]: Why is this `const` but everything else `let`?

File: cypress/integration/sGMR/report-form-validation.spec.js [hunk 4]
```diff
@@ -98,7 +99,11 @@ describe('Validate voyage plan form', () => {
         const errors2 = [
             'You must enter a registration number',
             'You must select if your pleasure craft has a country of registration',
-            'You must select if the pleasure craft has a call sign',
+        'You must select if the pleasure craft has an Automatic Identification System (AIS)',
```
Comment (p3dr0migue1) [inline cypress/integration/sGMR/report-form-validation.spec.js:102]: Indentation please

File: cypress/integration/sGMR/report-form-validation.spec.js [hunk 5]
```diff
@@ -158,15 +162,53 @@ describe('Validate voyage plan form', () => {
             cy.wrap(error).should('contain.text', errors2[index]).and('be.visible');
         });
     });
+    it('Should verify customs details mandatory data',()=>{
+        const errors =[
+            'Select if anyone onboard has goods in excess of their personal allowance',
+        'Select if there any prohibited or restricted goods on board',
```
Comment (p3dr0migue1) [inline cypress/integration/sGMR/report-form-validation.spec.js:168]: Please bear in mind indentation

File: cypress/integration/sGMR/report-form-validation.spec.js [hunk 6]
```diff
@@ -158,15 +162,53 @@ describe('Validate voyage plan form', () => {
             cy.wrap(error).should('contain.text', errors2[index]).and('be.visible');
         });
     });
+    it('Should verify customs details mandatory data',()=>{
```
Comment (p3dr0migue1) [inline cypress/integration/sGMR/report-form-validation.spec.js:165]: There should be a spaces around `() => {`

File: cypress/support/commands.js [hunk 7]
```diff
@@ -145,11 +154,18 @@ Cypress.Commands.add('enterArrivalDetails', (date, port) => {
   let dateTime = date.split(' ');
   const arrivalDate = dateTime[0].split('/');
   const arrivalTime = dateTime[1].split(':');
+  //const arrivalFromHour = (+arrivalTime[0]-3);
+  const arrivalFromHour = (+arrivalTime[0]);
+  const arrivalToHour = (arrivalFromHour +1);
+ // const arrivalToHour = (+arrivalTime[0]+1);
```
Comment (p3dr0migue1) [inline cypress/support/commands.js:160]: Please remove commented out code, not a good practice when we have version control

File: cypress/support/commands.js [hunk 8]
```diff
@@ -145,11 +154,18 @@ Cypress.Commands.add('enterArrivalDetails', (date, port) => {
...
   cy.get('input[name="arrivalDateDay"]').clear().type(arrivalDate[0]);
   cy.get('input[name="arrivalDateMonth"]').clear().type(arrivalDate[1]);
   cy.get('input[name="arrivalDateYear"]').clear().type(arrivalDate[2]);
-  cy.get('input[name="arrivalTimeFromHour"]').clear().type(arrivalTime[0]);
+//  cy.get('input[name="arrivalTimeFromHour"]').clear().type(arrivalTime[0]);
```
Comment (p3dr0migue1) [inline cypress/support/commands.js:164]: Please remove commented out code, not a good practice when we have version control

File: cypress/support/utils.js [hunk 9]
```diff
@@ -1,13 +1,20 @@
-exports.getFutureDate = (year, format) => {
+const dayjs = require("dayjs");
+/*exports.getFutureDate = (year, format) => {
   const dayjs = require('dayjs');
-  return ['DD/MM/YYYY', 'DD/MM/YYYY HH:mm', 'YYYY-MM-DD'].includes(format)
+  return ['DD/MM/YYYY', 'DD/MM/YYYY HH:mm', 'YYYY-MM-DD','HH:mm'].includes(format)
```
Comment (p3dr0migue1) [inline cypress/support/utils.js:4]: Please remove commented out code, not a good practice when we have version control

**Review by p3dr0migue1 — CHANGES_REQUESTED** [review]

**Comment by p3dr0migue1** [comment]
@JayanthiiB my preference would be to close this PR and create two new ones, where you make more than one explicit commit so that its easier for everyone to review. Please have a look at the Viable Data Coding Best Practices for a refresher.

File: cypress/integration/sGMR/add-people.spec.js (hunk 1 above)
Comment (JayanthiiB) [reply cypress/integration/sGMR/add-people.spec.js:39]: Hi @p3dr0migue1 ,thanks I have changed this in updated test as enterPeopleInfo and enterPeopleTravelDoc than keeping it as enterPeopleInfo and enterPeopleInfo1 in command.js to get values for People page.

File: cypress/integration/sGMR/add-vessel.spec.js (hunk 2 above)
Comment (JayanthiiB) [reply cypress/integration/sGMR/add-vessel.spec.js:42]: Thanks , Will be careful.

File: cypress/integration/sGMR/report-form-validation.spec.js (hunk 4 above)
Comment (JayanthiiB) [reply cypress/integration/sGMR/report-form-validation.spec.js:102]: Sure, thanks

File: cypress/integration/sGMR/report-form-validation.spec.js (hunk 5 above)
Comment (JayanthiiB) [reply cypress/integration/sGMR/report-form-validation.spec.js:168]: Sure , Thanks

File: cypress/integration/sGMR/report-form-validation.spec.js (hunk 6 above)
Comment (JayanthiiB) [reply cypress/integration/sGMR/report-form-validation.spec.js:165]: Sure, thanks .Have made the changes.

File: cypress/support/commands.js (hunk 8 above)
Comment (JayanthiiB) [reply cypress/support/commands.js:164]: Sure, Thanks

**Pull request closed without merging**